### ISSUE TITLE
Optionally hide channels the bot is in from 'network whois'

### ISSUE THREAD

**Issue by ghost** [ProgVal/Limnoria#98]
Issue sent from #limnoria at freenode by nyuszika7h (registered as nyuszika7h)

**Comment by GLolol** [comment]
Update: @Mikaela I can't reproduce that behavior. The bot hides `+s` channels when I use `whois` on my network.

**Comment by Mikaela** [comment]
@GLolol  Weird, I just tested by opening another client and connecting to IRCnet and running `/join 0`. Euforia says : 
> `[08:37:10] Euforia- Owner: Mikaela (mkaysi@lakka.kapsi.fi) has been on server irc.nebula.fi since 2014-12-20 19:10:05+0000 (idle for 16 hours, 5 minutes, and 28 seconds). euforia is an op on !channel.`

where !channel is +s. It appears to be hiding \#channels though (where it's also opped). The other bot appears to not reply to this connection at the moment.

**Comment by Mikaela** [comment]
I was going to open duplicate of this issue, but apparently this exists.

`@network whois` also doesn't care about mode +p which on freenode means that the channnel isn't shown in whois and on other network it does the same as +s and/or prevents /knock.

* http://freenode.net/using_the_network.shtml#modes

> p (paranoid) - When set, the KNOCK command cannot be used on the channel to request an invite, and **users will not be shown the channel in whois replies unless they are on it**. Unlike on some ircds, +p and +s can be set together.

**Comment by GLolol** [comment]
So basically, the bot should ignore `+p` channels just like it ignores `+s`? https://github.com/ProgVal/Limnoria/blob/testing/src/ircutils.py#L437

**Comment by Mikaela** [comment]
> So basically, the bot should ignore +p channels just like it ignores +s?

Yes. I can also see other issues here such as ! channels not being ignored, is # hardcoded somewhere?

**Comment by GLolol** [comment]
Does GLolol@3831cb762e34706e206d27fc136573d550137293 fix this issue then?

**Comment by Mikaela** [comment]
I have no interest in picking random commits outside of tagged releases, ask me again when it's there.

**Comment by GLolol** [comment]
I'm asking you to test the patch to see if it works :\
